src/ui/public/modals/modal_overlay.js

```diff
@@ -5,16 +5,23 @@ import modalOverlayTemplate from './modal_overlay.html';
  * Appends the modal to the dom on instantiation, and removes it when destroy is called.
  */
 export class ModalOverlay {
-  constructor(modalElement) {
+  constructor(modalElement, scope) {
     this.overlayElement = angular.element(modalOverlayTemplate);
     this.overlayElement.append(modalElement);
+
+    angular.element(document.body).on('keydown', (event) => {
+      if(event.keyCode === 27) {
+        scope.onCancel();
```

Comment:
If you both agree on that point I'd be happy to move that code down to the dialog level.